src/JobHelper.groovy

```diff
@@ -62,23 +62,53 @@ class JobHelper {
     /**
     * Queries an api and returns the results as a JSON object
     * @param query
+    * @param attempts
+    * @param context
     * @return
     */
-    private static queryApi(String query) {
-        try {
-            def get = new URL(query).openConnection()
-            get.setRequestProperty("User-Agent", "adopt-jenkins-helper")
-            def parser = new JsonSlurper()
-            return parser.parseText(get.getInputStream().getText())
-        } catch (Exception e) {
-            throw new RuntimeException("${e.getMessage()}")
+    private static queryApi(String query, Integer attempts, def context) {
```

Comment:
The library is pure groovy, as such println should be available without the context, and sleep can be achieved with:
```
try {
    Thread.sleep(30000);
} catch (Exception e) {
    println "Sleep interrupted"
}
```

If you do that you dont need the context

Reply:
I thought that too but when I tried it as part of my [first PR](https://github.com/AdoptOpenJDK/openjdk-jenkins-helper/pull/33/files), the println's that I put in didn't spit out any output https://ci.adoptopenjdk.net/job/build-scripts/job/openjdk16-pipeline/10/console